File: src/main/java/com/GoldenMine/object/Point2DCreator.java
package com.GoldenMine.object;

import com.GoldenMine.parser.Variable;

public class Point2DCreator implements KoreanObjectCreator {
    @Override
    public String getType() {
        return "xy좌표";
    }

    @Override
    public Variable.VariableMode getValueType(String key) {
        switch(key) {
            case "x":
            case "X":
            case "x좌표":
            case "X좌표":
                return Variable.VariableMode.REALNUM_MODE;
            case "y":
            case "Y":
            case "y좌표":
            case "Y좌표":
                return Variable.VariableMode.REALNUM_MODE;
        }

        return Variable.VariableMode.NULL;
    }

    @Override
    public String getRegex() {
        return "[d], [d]";
    }

    @Override
    public KoreanObject getInstance() {
        return new Point2D();
    }
}

File: src/main/java/kr/goldenmine/object/Point2DCreator.kt
package kr.goldenmine.`object`

import kr.goldenmine.parser.Variable

class Point2DCreator : KoreanObjectCreator {
    override val type: String
        get() = "xy좌표"

    override val regex: String
        get() = "[d], [d]"

    override val instance: KoreanObject
        get() = Point2D()

    override fun getValueType(key: String): Variable.VariableMode {
        when (key) {
            "x", "X", "x좌표", "X좌표" -> return Variable.VariableMode.REALNUM_MODE
            "y", "Y", "y좌표", "Y좌표" -> return Variable.VariableMode.REALNUM_MODE
        }

        return Variable.VariableMode.NULL
    }
}
